File: examples/stress_test.py
from force_directed_graph_qt import ForceDirectedGraph
from PyQt6.QtWidgets import QApplication

import sys
import random

app = QApplication(sys.argv)
graph = ForceDirectedGraph()

# setup window
graph.setWindowTitle("Node Graph")
graph.resize(800, 800)

# create nodes
ROWS = 20
COLUMNS = 20
GAP = 100

created_nodes = []

for row in range(ROWS):
    for column in range(COLUMNS):
        new_node = graph.createNode()
        new_node.setPosition(row*GAP, column*GAP)

        # connect node
        if(len(created_nodes)>0):
            index = random.randint(0, len(created_nodes)-1)
            connected_node = created_nodes[index]

            graph.connectNodes(new_node, connected_node)

        created_nodes.append(new_node)


# conections
# graph.connectNodes(node_a, node_b)

graph.show()
sys.exit(app.exec())

File: examples/stressTest/stressTest.cpp
#include <QApplication>
#include <QPushButton>
#include <cstdlib>

#include "ForceDirectedGraph.h"

int main(int argc, char **argv)
{
    QApplication app (argc, argv);

    int ROWS = 5;
    int COLUMNS = 5;
    int GAP = 20;

    fdg::ForceDirectedGraph* graph = new fdg::ForceDirectedGraph();

    std::vector<fdg::Node*> createdNodes;
    for(int row=0; row<ROWS; row++)
    {
        for(int column=0; column<COLUMNS; column++)
        {
            fdg::Node* newNode = graph->addNode("foo");
            newNode->setPos(row*GAP, column*GAP);

            // connect node
            if(createdNodes.size()>0)
            {
                int index = std::rand() % createdNodes.size();
                fdg::Node* connectedNode = createdNodes[index];

                graph->connectNodes(newNode, connectedNode);
            }

            createdNodes.push_back(newNode);
        }

    }


    graph->show();

    return app.exec();
}
